### PR TITLE
test(quality): Add missing ruff rules that are checked by flake8

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### Brief description of what is fixed or changed

Add rule F401 to ruff checks. Note that this rule is already checked by flake8 for the contents of the sympy code directory. The rule is basically `x is imported but not used` which means that a redundant import should be deleted. This is the most common reason that someone ends up pushing a second commit to fix the code quality CI checks but is currently not checked by running `ruff check`.

Adding this rule to ruff is useful because ruff can run the checks much faster than flake8 and can also fix these automatically most of the time. For example if I add an unnecessary import like:
```console
$ time ruff check 
sympy/core/add.py:12:17: F401 [*] `foo.bar` imported but unused
   |
10 | from .cache import cacheit
11 | from .numbers import equal_valued
12 | from foo import bar
   |                 ^^^ F401
13 | from .intfunc import ilcm, igcd
14 | from .expr import Expr
   |
   = help: Remove unused import: `foo.bar`

Found 1 error.
[*] 1 fixable with the `--fix` option.

real	0m0.159s
user	0m0.185s
sys	0m0.073s
```
Note that this checked the entire codebase in 0.2 seconds. It also says that it can fix the problem e.g.:
```console
$ ruff check --fix
Found 1 error (1 fixed, 0 remaining).
$ git diff
diff --git a/sympy/core/add.py b/sympy/core/add.py
index effbb4cd77..f09051ed8b 100644
--- a/sympy/core/add.py
+++ b/sympy/core/add.py
@@ -9,7 +9,6 @@
 from .operations import AssocOp, AssocOpDispatcher
 from .cache import cacheit
 from .numbers import equal_valued
-from foo import bar
 from .intfunc import ilcm, igcd
 from .expr import Expr
 from .kind import UndefinedKind
```
I don't usually run flake8 before pushing because it is too slow on the SymPy codebase. Running ruff is fast enough to be worthwhile.

#### Other comments

The ruff checks are set up to run in the whole codebase unlike the flake8 checks that only apply to the sympy package directory. Adding this rule to ruff checks means that it also applies to files outside the sympy package directory which is why I have removed some imports and added some noqa comments.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
In the last commit I enabled all ruff rules that are already checked by flake8. This should mean that ruff picks up on the same things that flake8 does and so we can remove the flake8 checks and just use ruff instead.

File: doc/src/conf.py [hunk 1]
```diff
@@ -99,7 +99,6 @@
     "modules/physics/mechanics/symsystem": "../explanation/modules/physics/mechanics/symsystem.html",
     "modules/physics/mechanics/linearize": "../explanation/modules/physics/mechanics/linearize.html",
     "modules/physics/mechanics/sympy_mechanics_for_autolev_uses": "../explanation/modules/physics/mechanics/sympy_mechanics_for_autolev_uses.html",
-    "modules/physics/mechanics/examples": "../tutorials/physics/mechanics.html",
```
Comment (moorepants) [inline doc/src/conf.py:102]: Is this mistakenly removed? I would guess we want to keep this redirect.

Comment (oscarbenjamin) [reply doc/src/conf.py:102]: There are two items with the same key in this dict. The ruff check looks for duplicate keys.

Comment (moorepants) [reply doc/src/conf.py:102]: OK

**Comment by haru-44** [comment]
looks good.

**Comment by oscarbenjamin** [comment]
Thanks.

In theory this should now mean that since the ruff check runs first in CI the flake8 check will never fail. We can let that run for a while to see if it is the case and then remove flake8 from CI checks and requirements-dev.txt as well as remove the .flake8 config file.

**Pull request merged**